org.jacoco.core/src/org/jacoco/core/internal/analysis/MethodCoverageCalculator.java

```diff
@@ -170,8 +188,28 @@ public void merge(AbstractInsnNode i1, AbstractInsnNode i2) {
 		}
 	}
 
+	/**
+	 * @deprecated use {@link #replaceBranches(AbstractInsnNode, Map)} instead
+	 */
+	@Deprecated
 	public void replaceBranches(final AbstractInsnNode source,
 			final Set<AbstractInsnNode> newTargets) {
+		final HashMap<AbstractInsnNode, Collection<InstructionBranch>> replacements = new HashMap<AbstractInsnNode, Collection<InstructionBranch>>();
+		for (final AbstractInsnNode target : newTargets) {
+			final ArrayList<InstructionBranch> list = new ArrayList<InstructionBranch>();
+			final int branches = Math.max(
```

Comment:
`Math.max` is used here for the same reason as in https://github.com/jacoco/jacoco/pull/1813/files#r1883024911